File: gdscript/src/main/java/gdscript/psi/GdPattern.java
package gdscript.psi;

import org.jetbrains.annotations.*;
import com.intellij.psi.PsiElement;

public interface GdPattern extends PsiElement {

  @Nullable
  GdArrayPattern getArrayPattern();

  @Nullable
  GdBindingPattern getBindingPattern();

  @Nullable
  GdDictPattern getDictPattern();

  @Nullable
  GdExpr getExpr();

}

File: gdscript/src/main/kotlin/gdscript/psi/GdPattern.kt
package gdscript.psi

import com.intellij.psi.PsiElement

interface GdPattern : PsiElement {
    val arrayPattern: GdArrayPattern?

    val bindingPattern: GdBindingPattern?

    val dictPattern: GdDictPattern?

    val expr: GdExpr?
}
